Moreschemachanges

bendk opened the pull request:
Several more schema changes to items.

I'd love some peer review on this one, but please don't merge until janet also tests it out.

Commit:
Rename item table column names to use underscore_names

I believe they only used camelCase before because miro used to be written in
that, and we never took the name to change the database schema.

Commit:
Refactored the logic for newly-available items.

Before we would set last_viewed on our feed, then use that to calculate if the
item was new (plus several other not entirely consistent criteria).  Changed
it so that we simply have a new column in the items table.

This is going to help the frontend data avoid needing to join to the feeds
table just to tell if an item is newly-available.

Commit:
Unroll remote_downloader.status column

Instead of storing a dictionary using repr, we store each value in its own
column.  This is cleaner in general and will help the frontend data project
since we need to be able to sort on some of these columns.

While we're messing with thing, I tried to refactor some of the logic for
these values.  Mostly I wanted to try to use None consistently to mean an
unknown or invalid value.  Also changed the type of some of the values.

  - eta/rate use None if we can't calculate them
  - endTime is None before the download has finishing, instead of being equal
    to startTime
  - totalSize is None if we don't know it instead of -1
  - retryCount uses None if we aren't retrying downloads instead of -1
  - Made filename/shortFilename initialize to None instead of ""
  - currentSize starts at 0 instead of None
  - downloaders don't send back the channelName value.  They never change it
    so that was just redundent.
  - Made startTime, endTime, eta, and rate integers instead of floats

Changed the hack for torrent magnet link titles.  Instead of setting
item.title, it sets torrent_title.  It also uses the metadata rather than the
filename attribute.  I think this is more correct than before.

Commit:
Renamed downloader column to match PEP 8.

This doesn't change the semantics of anything, just the names.

Changing an existing database upgrade is a bit dirty, but I figure it's okay
since this hasn't been merged to master yet.

Commit:
Fixing fallout.

Fixed a bunch of small issues based on janet doing a pass through the code.

Commit:
Another quick fix after the schema changes

Commit:
More bugfixes

Commit:
Moving setup_restored to be closer to setup_new()

Commit:
Initialize variables to None

We used to use special int values for these, but None is the new value for
unknown.

Commit:
Don't store temporary values in the DB.

There's no reason to store these in the DB since we just reset them when we
start up again.

Commit:
Re-implementing the _save_later() hack in downloader.py

Commit:
Use lowercase for column types.

We haven't been consistent with this always, but I think it makes sense to
use lowercase.  That's what we use when we create new tables.

Commit:
fix ALTER TABLE syntax to match what we use in other places

Commit:
Rearanged RemoteDownloader.update_status a bit.

Commit:
Drop the seen column on item.

It's just redundent when we have watched_time.

Dropped references to "seen" and changed them to "watched".  It's what we use
in the frontend and it's more clear that "watched" means the file has been
played, rather then the user has just seen the item in an item list.

Fixed typo that accidentally worked in upgrade182.

bendk commented:
Janet has tested most of this (minus the last 4 commits or so).  I think that's good enough to merge if it passes peer-review.

paulswartz commented on tv/lib/databaseupgrade.py:
```diff
@@ -3828,3 +3828,124 @@ def get_title(metadata_title, torrent_title, entry_title, filename):
+    # Rename columns in the item table
+    rename_columns = {
+        'autoDownloaded': 'auto_downloaded',
+        'pendingManualDL': 'pending_manual_dl',
```
If we're going to rename it anyways, why not `pending_manual_download`?

paulswartz commented on tv/lib/databaseupgrade.py:
```diff
@@ -3828,3 +3828,124 @@ def get_title(metadata_title, torrent_title, entry_title, filename):
+    cursor.execute("ALTER TABLE remote_downloader "
+                   "ADD COLUMN short_reason_failed text")
+    cursor.execute("ALTER TABLE remote_downloader "
+                   "ADD COLUMN dler_type text")
```
`downloader_type`?

Commit:
Rename "pending_manual_dl" to "pending_manual_download"

Paul's right.  As long as we're renaming, we may as well pick a good name.

Commit:
Another z3p suggestion, don't use the awkward name dler_type.

Replaced it with simply "type".

Also fixed upgrade 182.  We weren't correctly handling the fact the attribute
names and the keys in the old status dict were different.

bendk commented:
Added some commits based on the review.  If it looks good, I think we can merge.  Janet's tested it.
